workspaces/cli-server/src/tasks/on-demand-initial-body.ts

```diff
@@ -72,7 +72,7 @@ export class OnDemandInitialBody {
     };
     const onExit = (code: number, signal: string | null) => {
       cleanup();
-      if (code !== 0) {
+      if (code !== 0 && code !== null) {
```

Comment:
this function should be updated to have the correct argument types, if null is possible for `code`. It should check and log the signal argument or exit code. This function is duplicated in each of the tasks so those should be updated as well.